.github/workflows/interface-unit-tests.yml

```diff
@@ -66,6 +71,22 @@ jobs:
             "default": ["3.10"]
           }
           EOF
+          elif [ "${{ inputs.python_warning_level }}" == "error" ];
+          then
+            cat >python_versions.json <<-EOF
+          {
+            "default": ["3.10"],
+            "torch-tests": ["3.10"],
+            "tf-tests": ["3.10"],
+            "jax-tests": ["3.10"],
+            "all-interfaces-tests": ["3.10"],
+            "external-libraries-tests": ["3.10"],
+            "qcut-tests": ["3.10"],
+            "qchem-tests": ["3.10"],
+            "gradients-tests": ["3.10"],
+            "data-tests": ["3.10"],
+            "device-tests": ["3.10"]          }
+          EOF
```

Comment:
Couldn't this entire branch be combined with the one for lightened CI? Since all the runners will use 3.10 anyway in both cases.

Reply:
I wanted to use 3.12 for this, as it is a better indicator of what Python is removing, but don't have the time to update the dependency versions for those without a 3.12 wheel. I'd like to keep this as it will be updated for that purpose.

Reply:
In that case you can still get rid of everything except for the `default`, and we can eventually replace that with 3.12.